packages/broker/test/integration/plugins/operator/OperatorPlugin.test.ts

```diff
@@ -1,29 +1,84 @@
-import { fastWallet } from '@streamr/test-utils'
+import type { Operator } from '@streamr/network-contracts'
+import { fastPrivateKey, fetchPrivateKeyWithGas } from '@streamr/test-utils'
+import { waitForCondition } from '@streamr/utils'
 import { Wallet } from 'ethers'
-import { Broker } from '../../../../src/broker'
-import { startBroker } from '../../../utils'
+import { ProxyDirection, StreamPermission } from 'streamr-client'
+import { Broker, createBroker } from '../../../../src/broker'
+import { createClient, createTestStream, formConfig, startBroker } from '../../../utils'
+import { delegate, deploySponsorshipContract, generateWalletWithGasAndTokens, setupOperatorContract, sponsor, stake } from './contractUtils'
 
 describe('OperatorPlugin', () => {
 
-    let brokerWallet: Wallet
     let broker: Broker
+    let brokerWallet: Wallet
+    let operatorContract: Operator
+    let operatorWallet: Wallet
 
-    beforeEach(async () => {
-        brokerWallet = fastWallet()
-    })
+    beforeAll(async () => {
+        const deployment = (await setupOperatorContract({
+            nodeCount: 1
+        }))
+        brokerWallet = deployment.nodeWallets[0]
+        operatorWallet = deployment.operatorWallet
+        operatorContract = deployment.operatorContract
+    }, 30 * 1000)
 
     afterEach(async () => {
-        broker?.stop()
+        await broker?.stop()
     })
 
-    it('can start broker with operator plugin', async () => {
+    it('accepts proxy connections', async () => {
+        const subscriber = createClient(await fetchPrivateKeyWithGas())
+        const stream = await createTestStream(subscriber, module)
+
+        const sponsorer = await generateWalletWithGasAndTokens()
```

Comment:
I would do all sponsoring related stuff first and then go to delegating and staking